packages/core/src/shared/sam/localLambdaRunner.ts

```diff
@@ -236,7 +233,7 @@ async function invokeLambdaHandler(
             templatePath: config.templatePath,
             eventPath: config.eventPayloadFile,
             environmentVariablePath: config.envFile,
-            environmentVariables: await getSpawnEnv(env),
```

Comment:
This solution applies the same logic to lower-tier versions that we currently use for customer downgrades. I am still skeptical about this solution but i think we are on the right track

Reply:
The error happens cause `getSpawnEnv` will mistakenly output a overwritten process.env.PATH on Windows before this patch. And in build process in local invoke, this will overwrite the process.env.PATH i. Resulting sam build can't find the binary of npm/dotnet. After this Patch, getSpawnEnv(env) will directly return env correctly if it's on windows.

For lower version, process.env.PATH will be null in the envVar. And it won't overwrite process.env.PATH in later part.

For Mac, `getSpawnEnv` will output the right path so there's no issue on Mac